07_modules_libs/homework/stockserver.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+// result of successful query
+type jsonRes struct {
+	Ticker      string      `json:"ticker"`
+	Date        string      `json:"date"`
+	Open_price  interface{} `json:"open_price"`
+	High_price  interface{} `json:"high_price"`
+	Low_price   interface{} `json:"low_price"`
+	Close_price interface{} `json:"close_price"`
+}
+
+// result of the query
+// if query is successful result will be placed in "value"
+// or you will get error
+type result struct {
+	value jsonRes
+	err   error
+}
+
+type customError struct {
+	Error string
+}
+
+//handle func for endpoint /price/{ticker}/date/{date}
+func (s StockServer) price(w http.ResponseWriter, r *http.Request) {
+
+	ticker := chi.URLParam(r, "ticker")
+	date := chi.URLParam(r, "date")
+
+	ctx := r.Context()
+	readyCh := make(chan result)
+
+	go s.avClient.pricesByDate(ticker, date, readyCh)
+
+	select {
+	case answer := <-readyCh:
+		if answer.err != nil {
+			writeResponse(w, http.StatusBadRequest, customError{answer.err.Error()})
+			return
+		}
+		writeResponse(w, http.StatusOK, answer.value)
+	case <-ctx.Done():
+		log.Printf("Interrupted with err: %v", ctx.Err())
+	}
+}
+
+func writeResponse(w http.ResponseWriter, code int, v interface{}) {
+	b, _ := json.Marshal(v)
+	w.WriteHeader(code)
+	w.Write([]byte(b))
+}
+
+//run StockServer
+func (server *StockServer) Run(port interface{}) {
+
+	var strPort string
+	switch port.(type) {
```

Comment:
You can just use `fmt.Sprintf(":%v",port)` to get string representation of port. Keep in my that we should rid of using `interfaces{}`.